client/app/telecom/pack/xdsl/access/diagnostic/details/pack-xdl-access-diagnostic-details.controller.js

```diff
@@ -0,0 +1,44 @@
+angular.module('managerApp').controller('XdslDiagnosticDetailsCtrl', class XdslDiagnosticDetailsCtrl {
+  constructor($scope, $rootScope, $state, $translate) {
+    this.$scope = $scope;
+    this.$rootScope = $rootScope;
+    this.$state = $state; // used in html
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    this.diagnostic = null;
+    this.statuses = null;
+
+    this.$scope.$on('accessDiagnosticDetails:arrived', (event, diagnostic) => {
+      this.diagnostic = diagnostic;
+      this.statuses = this.constructor.getStatuses(diagnostic || {});
+    });
+
+    this.$rootScope.$emit('accessDiagnosticDetails:get');
+  }
+
+  reload() {
+    this.diagnostic = null;
+    this.$rootScope.$emit('accessDiagnosticDetails:launch');
+  }
+
+  static getStatuses(diagnostic) {
+    return Object.entries(_.pick(diagnostic, ['isActiveOnLns', 'ping', 'isModemConnected']))
+      .map(kv => ({ key: kv[0], val: !!kv[1] }));
+  }
+
+  getRemainingChecksText() {
+    const checkText = this.$translate.instant('xdsl_details_diagnostic_modal_relaunch_status_check');
+
+    if (this.diagnostic && this.diagnostic.remaining) {
```

Comment:
```suggestion
    if (_.has(this.diagnostic,'remaining')) {
```
?

Reply:
I think I prefer the symmetry with the other line further down.